File: LeetCode/U_202_HappyNumber.py
# https://leetcode.com/problems/happy-number/
# 2022年04月22日 14:59:28

class Solution:
    def isHappy(self, n: int) -> bool:
        q=[]
        ans=0
        while ans!=1:
            ans=0
            while n!=0:
                ans+=(n%10)**2
                n=(n-n%10)//10
            n=ans
            if ans not in q:
                q.append(ans)
            else:
                return False
            # print(ans)
        return True

File: LeetCode_rename/U_202_HappyNumber.go
// https://leetcode.com/problems/happy-number/submissions/
// 2022年04月22日 15:52:53

func isHappy(n int) bool {
    var q []int
    ans:=0
    for ;ans!=1; {
        ans=0
        for ;n!=0; {
            ans+=int(math.Pow(float64(n%10),2))
            n=(n-n%10)/10
        }
        n=ans
        // fmt.Println(ans)
        for i:=0;i<len(q);i++ {
            if q[i]==ans {
                // fmt.Println("same found")
                return false
            }
        }
        q=append(q,ans)
    }
    return true
}